homeassistant/components/suez_water/__init__.py

```diff
@@ -26,3 +32,47 @@ async def async_setup_entry(hass: HomeAssistant, entry: SuezWaterConfigEntry) ->
 async def async_unload_entry(hass: HomeAssistant, entry: SuezWaterConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(
+    hass: HomeAssistant, config_entry: SuezWaterConfigEntry
+) -> bool:
+    """Migrate old suez water config entry."""
+    _LOGGER.debug(
+        "Migrating configuration from version %s.%s",
+        config_entry.version,
+        config_entry.minor_version,
+    )
+
+    if config_entry.version > SuezWaterConfigFlow.VERSION:
+        _LOGGER.error("Suez_water version downgrade not handled")
+        return False
+    if config_entry.version == 2:
+        _LOGGER.debug("Suez_water no minor changes in version 2.X")
+        return True
+
+    unique_id = config_entry.unique_id
+    if config_entry.version == 1:
+        # Going to 2.X
+        counter_id = config_entry.data.get(CONF_COUNTER_ID)
+        if not counter_id:
```

Comment:
Is this possible?

Reply:
No because the integration doesn't  work if it is missing.
But I preferred to check in any case just to be sure 😅

Reply:
Then we need to have test coverage for something that can't happen so I suggest you'd remove it.

Reply:
it's gone